### DISCUSSION TITLE
play_sound possible inside gcode_macro

### DISCUSSION THREAD

**Discussion by TheLexus** [Desuuuu/klipper#108] [Q&A] [answered]
Hi,

first thanks for all the time you invest and invested in this fork, works very well on my E6 (im on the latest version with customizable menus...). 

Is there any way to call the play_sound function from a gcode_macro (in my case playing the jingle at print end)?

Bye!

**Comment by Desuuuu** [comment, marked as answer]
Sure, there are gcode commands to do just that: `DGUS_PLAY_SOUND`/`M300`

For the `dgus-display` branch, they're documented [here](https://github.com/Desuuuu/klipper/blob/dgus-display/docs/G-Codes.md).

They might work slightly differently on the `dgus-reloaded` branch so you might want to take a look at the [source](https://github.com/Desuuuu/klipper/blob/dgus-reloaded/klippy/extras/t5uid1/t5uid1.py#L871) for the details.

**Comment by TheLexus** [comment]
Perfect, thanks a lot.
